types/src/client.rs

```diff
@@ -16,41 +16,50 @@ pub struct Subscription<Notif> {
 	pub marker: PhantomData<Notif>,
 }
 
+/// Notification.
+#[derive(Debug)]
+pub struct NotificationMessage {
+	/// Method for the notification.
+	pub method: String,
+	/// Parameters to send to the server.
+	pub params: Params,
+}
+
+/// Request
+#[derive(Debug)]
+pub struct RequestMessage {
+	/// Method for the request.
+	pub method: String,
+	/// Parameters of the request.
+	pub params: Params,
+	/// One-shot channel over which we send back the result of this request.
+	pub send_back: Option<oneshot::Sender<Result<JsonValue, Error>>>,
+}
+
+/// Subscribe.
+#[derive(Debug)]
+pub struct SubscriptionMessage {
+	/// Method for the subscription request.
+	pub subscribe_method: String,
+	/// Parameters to send for the subscription.
+	pub params: Params,
+	/// Method to use to unsubscribe later. Used if the channel unexpectedly closes.
+	pub unsubscribe_method: String,
+	/// When we get a response from the server about that subscription, we send the result on
+	/// this channel. If the subscription succeeds, we return a [Receiver](futures::channel::mpsc::Receiver) that will receive
+	/// notifications.
+	pub send_back: oneshot::Sender<Result<(mpsc::Receiver<JsonValue>, SubscriptionId), Error>>,
+}
+
 /// Message that the Client can send to the background task.
 #[derive(Debug)]
 pub enum FrontToBack {
 	/// Send a one-shot notification to the server. The server doesn't give back any feedback.
```

Comment:
I think we should stray the last comment because I think it's rather implementation specific and the `JSONRPC spec` does state anything about it:

> A Request object that is a Notification signifies the Client's lack of interest in the corresponding Response object, and as such no Response object needs to be returned to the client.

Reply:
For WS it's fine to not respond but for HTTP `200 OK` seems reasonable for a server to respond with see #212